crates/bundler/src/bundler.rs

```diff
@@ -1,40 +1,151 @@
+    types::{
+        transaction::eip2718::TypedTransaction, Address, Eip1559TransactionRequest, H256, U64,
+    },
+};
+use ethers_flashbots::{
+    BundleRequest, FlashbotsMiddleware, PendingBundleError::BundleNotIncluded, SimulatedBundle,
 };
 use silius_contracts::entry_point::EntryPointAPI;
-use silius_primitives::{Chain, UserOperation, Wallet};
+use silius_primitives::{consts::flashbots_relay_endpoints, Chain, UserOperation, Wallet};
 use std::{sync::Arc, time::Duration};
 use tracing::{info, trace};
+use url::Url;
+
+/// A type alias for the Flashbots Signer client
+type FlashbotsClientType<M> =
+    Arc<SignerMiddleware<FlashbotsMiddleware<M, LocalWallet>, LocalWallet>>;
 
-#[derive(Clone)]
+/// A type alias for the Ethereum Signer client
+type EthClientType<M> = Arc<SignerMiddleware<M, LocalWallet>>;
+
+/// The `SendBundleMode` determines whether to send the bundle to a Ethereum execution client or to Flashbots relay
+#[derive(Clone, Debug, PartialEq)]
+pub enum SendBundleMode {
+    /// Send the bundle to a Ethereum execution client
+    EthClient,
+    /// Send the bundle to Flashbots relay
+    Flashbots,
+}
+
+/// The `Bundler` struct is used to represent a bundler with necessary properties
+#[derive(Clone, Debug)]
 pub struct Bundler {
+    /// Wallet instance representing the bundler's wallet.
     pub wallet: Wallet,
+    /// URL of an Ethereum client.
     pub eth_client_address: String,
+    /// Beneficiary address where the gas is refunded after execution
     pub beneficiary: Address,
+    /// Entry point contract address
     pub entry_point: Address,
+    /// [Chain](Chain) instance representing the blockchain network to be used
     pub chain: Chain,
+    /// Send bundle mode determines whether to send the bundle to a regular Eth execution client or to Flashbots
+    pub send_bundle_mode: SendBundleMode,
+    /// Block Builder relay endpoints
+    pub relay_endpoints: Option<Vec<String>>,
+    /// Ethereum signer middleware
+    pub eth_client: Option<EthClientType<Provider<Http>>>,
+    /// Flashbots signer middleware
+    pub fb_client: Option<FlashbotsClientType<Provider<Http>>>,
 }
 
 impl Bundler {
+    /// Create a new `Bundler` instance
+    /// if `send_bundle_mode` is `SendBundleMode::Flashbots` and `relay_endpoints` is `None`, the default Flashbots relay endpoint will be used
+    ///
+    /// # Returns
+    /// * `Self` - A new `Bundler` instance
     pub fn new(
         wallet: Wallet,
         eth_client_address: String,
         beneficiary: Address,
         entry_point: Address,
         chain: Chain,
-    ) -> Self {
-        Self {
-            wallet,
-            eth_client_address,
-            beneficiary,
-            entry_point,
-            chain,
+        send_bundle_mode: SendBundleMode,
+        relay_endpoints: Option<Vec<String>>,
+    ) -> anyhow::Result<Self> {
+        if !(chain.id() == 1 || chain.id() == 5 || chain.id() == 11155111)
```

Comment:
By network id, do you mean chain id? Chain id is already included in the Bundler struct, and we can check which chain the bundler is on by `chain::id()`

Can you maybe give a pseudo code example or the context where this will be necessary? 

Thanks!

Reply:
Option 1:

`chain.id() == ethers::types::Chain::Sepolia`

Option 2:

`chain::name() == "sepolia"`

I think it's better to have it like that because it's easier later to search in the code and find all specifics for some chain (for example gas things will be different for chain as well). For example if I search mainnet, I will find exactly things related to mainnet, but if I search "1" it will be a mess.